scripts/src/lib/template/templates/gradle/kotlin/build.gradle.kts.eta

```diff
@@ -0,0 +1,113 @@
+plugins {
+	id("fabric-loom") version "1.0-SNAPSHOT"
+	id("maven-publish")<% if (it.kotlin) { %>
+	kotlin("jvm") version "<%= it.kotlin.kotlinVersion %>"<% } %>
+}
+
+java.sourceCompatibility = JavaVersion.<%= it.java.compatibility %>
+java.targetCompatibility = JavaVersion.<%= it.java.compatibility %>
+
+base.archivesName.set(project.properties["archives_base_name"] as String)
+version = project.properties["mod_version"] as String
+group = project.properties["maven_group"] as String
+
+repositories {
+	// Add repositories to retrieve artifacts from in here.
+	// You should only use this when depending on other mods because
+	// Loom adds the essential maven repositories to download Minecraft and libraries from automatically.
+	// See https://docs.gradle.org/current/userguide/declaring_repositories.html
+	// for more information about repositories.
+}
+<% if (it.dataGeneration || it.splitSources) { %>
+loom {
+<% if (it.splitSources) { %>	splitEnvironmentSourceSets()
+
+	mods {
+		create("modid") {
+			sourceSet(sourceSets["main"])
+			sourceSet(sourceSets["client"])
+		}
+	}    
+<% } %><% if (it.dataGeneration) { %>	runs {
+		create("datagen") {
+			inherit(runConfigs["server"])
+			name("Data Generation")
+			vmArg("-Dfabric-api.datagen")
+			vmArg("-Dfabric-api.datagen.output-dir=${file("src/main/generated")}")
+			vmArg("-Dfabric-api.datagen.modid=<%= it.modid %>")
+
+			runDir("build/datagen")
+		}
+	}<% } %>
+}
+<% } %><% if (it.dataGeneration) { %>
+sourceSets {
+	main {
+		resources {
+			srcDirs += file("src/main/generated")
+		}
+	}
+}<% } %>
+
+dependencies {
+	// To change the versions see the gradle.properties file
+	minecraft("com.mojang:minecraft:${project.properties["minecraft_version"]}")
+	mappings("net.fabricmc:yarn:${project.properties["yarn_mappings"]}:v2")
+	modImplementation("net.fabricmc:fabric-loader:${project.properties["loader_version"]}")
+
+	// Fabric API. This is technically optional, but you probably want it anyway.
+	modImplementation("net.fabricmc.fabric-api:fabric-api:${project.properties["fabric_version"]}")
+<% if (it.kotlin) { %>	modImplementation("net.fabricmc:fabric-language-kotlin:${project.properties["fabric_kotlin_version"]}")<% } %>
+	// Uncomment the following line to enable the deprecated Fabric API modules. 
+	// These are included in the Fabric API production distribution and allow you to update your mod to the latest modules at a later more convenient time.
+
+	// modImplementation("net.fabricmc.fabric-api:fabric-api-deprecated:${project.properties["fabric_version"]}")
+}
+
+tasks {
+	processResources {
+		inputs.property("version", project.version)
+
+		filesMatching("fabric.mod.json") {
+			expand(mapOf("version" to project.version))
+		}
+	}
+
+	withType<JavaCompile> {
+		options.release.set(java.targetCompatibility.majorVersion.toInt())
```

Comment:
Shouldn't this be `<%= it.java.release %>` like in the groovy template?

Reply:
I wouldn't create more constants, This PR is about Kotlin DSL, but I think, it should be changed in groovy to reuse the `targetCompatibility` constant

Reply:
IMO the best solution would be to use top-level constants in both Kotlin and Groovy buildscripts (`val`/`def` respectively) for these constants.

Since "this PR is about Kotlin DSL", shouldn't it follow the existing pattern?